tests/test_cli_compile.py

```diff
@@ -273,6 +273,35 @@ def test_locally_available_editable_package_is_not_archived_in_cache_dir(
     assert not os.listdir(os.path.join(str(cache_dir), "pkgs"))
 
 
+def test_url_package(runner):
+    url_package = (
+        # use version prior to pip-tools use of setuptools_scm,
+        # which is incompatible with url install
+        "https://github.com/jazzband/pip-tools/archive/"
+        "7d86c8d3ecd1faa6be11c7ddc6b29a30ffd1dae3.zip#egg=pip-tools"
+    )
+    with open("requirements.in", "w") as req_in:
+        req_in.write(url_package)
+    out = runner.invoke(cli, ["-n", "--rebuild"])
+    assert out.exit_code == 0
+    assert url_package in out.output
+    assert "click" in out.output  # dependency of pip-tools
+
+
+def test_url_package_vcs(runner):
+    vcs_package = (
+        "git+git://github.com/pytest-dev/pytest-django"
+        "@21492afc88a19d4ca01cd0ac392a5325b14f95c7"
+        "#egg=pytest-django"
+    )
+    with open("requirements.in", "w") as req_in:
+        req_in.write("-e " + vcs_package)
+    out = runner.invoke(cli, ["-n", "--rebuild"])
+    assert out.exit_code == 0
+    assert vcs_package in out.output
+    assert "pytest" in out.output  # dependency of pytest-django
+
+
```

Comment:
Could you add a test for a package with a link something like`https://files.pythonhosted.org/.../pip-8.1.1-py2.py3-none-any.whl`